Templates/JenkinsPipeline/Jenkinsfile

```diff
@@ -501,16 +571,31 @@ pipeline {
 							sh "chtag -c UTF-8 shiplist-utf-8.xml"
 						}
 						
-						println("${PipelineName}[INFO]: Gathering UCD Component Version URL.")
+						println("${PipelineName}[INFO]: Gathering UCD Component Version Name and URL.")
+
+						// Locate the UCD Component Version Name in the output log.
+						def ucdVersionNamePattern = java.util.regex.Pattern.compile(".*Computed UCD component version name: (.*)")
+						def ucdVersionMatcher = ucdVersionNamePattern.matcher(readFile("${ucdBuzToolLog}"))
+						if (ucdVersionMatcher.find()) {
+							ucdVersionName = "${ucdVersionMatcher.group(1)}"
+							ucdVersionMatcher = null // Prevent java.io.NotSerializableException: java.util.regex.Matcher Exception.
```

Comment:
That line feels a bit strange. Shouldn't it be managed with a try/catch block instead?

Reply:
> That line feels a bit strange. Shouldn't it be managed with a try/catch block instead?

@M-DLB I have reworked this section and externalised the method of the non-seriablizable call with the regex matcher. Additionally, I have improved the management of the **stage results**. 

Can you please review the recent commit https://github.com/IBM/dbb/pull/245/commits/0f3bbd4dc6ea25e3297d1ae7cca4878eb6b92278